java-checks/src/main/java/org/sonar/java/symexecengine/ExecutionState.java

```diff
@@ -106,10 +119,12 @@ public ExecutionState restoreParent() {
     return parent.merge(this);
   }
 
+  @Deprecated
   Set<State> getStatesOfCurrentExecutionState() {
     Set<State> results = Sets.newHashSet();
     for (Symbol symbol : definedInState) {
-      for (SymbolicValue value : Iterables.concat(reachableValues.get(symbol), unreachableValues.get(symbol))) {
+      // FIXME(merciesa): why iterating over unreachableValues ?
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png) ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)